pulp_smash/tests/pulp3/pulpcore/api_v3/test_status.py

```diff
@@ -21,35 +58,41 @@ class StatusTestCase(unittest.TestCase, utils.SmokeTest):
     * `Pulp Smash #755 <https://github.com/PulpQE/pulp-smash/issues/755>`_
     """
 
-    @classmethod
-    def setUpClass(cls):
-        """Create class-wide variables."""
-        cls.cfg = config.get_config()
-        cls.client = api.Client(cls.cfg, api.json_handler)
-        del cls.client.request_kwargs['auth']
-        cls.status = {}
-
-    def test_01_access_status(self):
-        """Verify whether an un-authenticated user can view status page."""
-        self.status.update(self.client.get(STATUS_PATH))
-
-    @selectors.skip_if(bool, 'status', False)
-    def test_02_data_content(self):
-        """Verify whether few parameters are present on status page."""
-        self.assertTrue(self.status['database_connection'])
-        self.assertTrue(self.status['messaging_connection'])
-        self.assertIsInstance(self.status['online_workers'], list)
-        self.assertIsInstance(self.status['missing_workers'], list)
-        self.assertIsInstance(self.status['versions'], list)
-        self.assertNotEqual(self.status['online_workers'], [])
-        self.assertEqual(self.status['missing_workers'], [])
-        self.assertNotEqual(self.status['versions'], [])
-
-    def test_03_http_method(self):
-        """Verify whether an HTTP exception is raised.
-
-        When using a not allowed HTTP method.
+    def setUp(self):
+        """Make an API client."""
+        self.client = api.Client(config.get_config(), api.json_handler)
+
+    def test_get_authenticated(self):
+        """GET the status path with valid credentials.
+
+        Verify the response with :meth:`verify_get_response`.
+        """
+        self.verify_get_response(self.client.get(STATUS_PATH))
+
+    def test_get_unauthenticated(self):
+        """GET the status path with no credentials.
+
+        Verify the response with :meth:`verify_get_response`.
+        """
+        del self.client.request_kwargs['auth']
+        self.verify_get_response(self.client.get(STATUS_PATH))
+
+    def test_post_authenticated(self):
+        """POST the status path with valid credentials.
+
+        Assert an error is returned.
         """
-        self.client.request_kwargs['auth'] = get_auth()
         with self.assertRaises(HTTPError):
             self.client.post(STATUS_PATH)
+
+    def verify_get_response(self, status):
+        """Verify the response to an HTTP GET call.
+
+        Verify that several attributes and have the correct type or value.
+        """
+        validate(status, STATUS)
+        self.assertTrue(status['database_connection']['connected'])
```

Comment:
This is an example of an assertion that's being made more strict *without* making use of a schema. Formerly, this assertion read as follows:

```python
self.assertTrue(status['database_connection'])
```

Reply:
@Ichimonji10, pretty neat. Thanks for updating this.